Add support for IAST SQL injection system tests

manuel-alvarez-alvarez opened the pull request:
## Description

Add system tests for IAST SQL injection detection

robertomonteromiguel commented on tests/appsec/iast/test_iast.py:
```diff
@@ -56,3 +67,19 @@ def test_insecure_md5_hash(self):
 
         interfaces.library.expect_iast_vulnerabilities(r, vulnerability_type="WEAK_HASH", evidence="md5")
+
+    def test_secure_sql(self):
```
@missing_feature(library="nodejs", reason="Need to implement nodejs endpoint")

manuel-alvarez-alvarez replied:
Changed, thanks!

robertomonteromiguel commented on utils/build/docker/java/spring-boot/pom.xml:
```diff
@@ -89,7 +93,15 @@
 		    <version>1.70</version>
 		</dependency>
-
+        <dependency>
```
do we really need flyway?

manuel-alvarez-alvarez replied:
Nope, not really, the same can be done with spring, removed!

cbeauchesne commented:
Hi @manuel-alvarez-alvarez , reading from the conflict, I think what you tried to test isnow in the main branch. Can I close this PR ?

manuel-alvarez-alvarez commented:
> Hi @manuel-alvarez-alvarez , reading from the conflict, I think what you tried to test isnow in the main branch. Can I close this PR ?

Hello @cbeauchesne the functionality this test should cover is not yet merged (at least not in the java library, [Add SQL Injection detection #3909](https://github.com/DataDog/dd-trace-java/pull/3909)), but it's taking forever :cry: 

Should and I mark the new tests as missing for all libraries so we can merge the PR?

PD: Sorry for the delay in the response, I got to improve my filtering of github notifications.

cbeauchesne commented:
> Should I mark the new tests as missing for all libraries so we can merge the PR?

Yes, simply uses `@missing_feature`. The good part is that,  on your PR, you'll see it as x-passed if it's ok. Once your PR is merged on java tracer, you will do a simple PR  here to remove the decorator for java.

manuel-alvarez-alvarez commented:
> > Should I mark the new tests as missing for all libraries so we can merge the PR?
> 
> Yes, simply uses `@missing_feature`. The good part is that, on your PR, you'll see it as x-passed if it's ok. Once your PR is merged on java tracer, you will do a simple PR here to remove the decorator for java.

Thanks for the input!, I've rebased from main and fixed the conflicts so it should be ready for the review and merge :smile:
